### PR TITLE
add a z:current filter option

### PR DESCRIPTION
**Pull request by opaldes**
#309

### CODE REVIEW

**Comment by NoahTheDuke** [comment]
Thanks for this! I'll test it out and see!

File: src/AppBundle/Service/RotationService.php [hunk 1]
```diff
@@ -22,6 +22,22 @@ public function __construct(EntityManagerInterface $entityManager)
         $this->entityManager = $entityManager;
     }
 
+    /**
+     * [returns the first entry of the descending sorted rotations]
+     * @return Rotation current Rotation
+     */
+    public function findCurrentRotation()
+    {
+        $rotations = $this->entityManager->getRepository(Rotation::class)->findBy([], ['dateStart' => 'DESC']);
```
Comment (plural) [inline src/AppBundle/Service/RotationService.php:31]: you could also consider

$rotation = $this->entityManager->getRepository(Rotation::class)->findOneBy([], ['dateStart' => 'DESC']);
if (!$rotation) {
  throw ...
}
return $rotation;

**Review by plural — CHANGES_REQUESTED** [review]
This is excellent!  I patched it in and it works well.

Please also update the syntax page at app/Resources/views/Default/syntax.html.twig and the tooltip for syntax at app/Resources/views/Search/searchtooltip.html.twig

(i should pull those together so we don't have them duplicated...)

Another option we could consider here is expanding this just a little bit:

z:rotation#|current|latest

the rotations have names and codes in the DB right now, but AFAIK, they aren't shown anywhere.

mysql> SELECT * FROM rotation;
+----+---------------+-----------------+---------------------+---------------------+------------+
| id | code          | name            | date_creation       | date_update         | date_start |
+----+---------------+-----------------+---------------------+---------------------+------------+
|  1 | rotation-2017 | First Rotation  | 2019-12-26 22:29:02 | 2019-12-26 22:29:02 | 2017-10-01 |
|  2 | rotation-2018 | Second Rotation | 2019-12-26 22:29:02 | 2019-12-26 22:29:02 | 2018-12-21 |
|  3 | rotation-2019 | Third Rotation  | 2019-12-26 22:29:02 | 2019-12-26 22:29:02 | 2019-12-27 |
+----+---------------+-----------------+---------------------+---------------------+------------+

File: src/AppBundle/Service/RotationService.php (hunk 1 above)
Comment (opaldes) [reply src/AppBundle/Service/RotationService.php:31]: will do seems better because we expect only 1 to return

Comment (jeremy-seipelt) [reply src/AppBundle/Service/RotationService.php:31]: @plural do we want z:"First Rotation" or z:rotation-2017 ? I would change my function appropriately

Comment (plural) [reply src/AppBundle/Service/RotationService.php:31]: no quoting is better, i think, so i would go with rotation-2017.  We can always publish the names of the valid rotation strings on the syntax page to help people out, too.

**Comment by plural** [comment]
hey @opaldes, do you want to push ahead with this or have one of us run with it?  Thanks for kicking this off.

**Comment by plural** [comment]
i ran with this in #427 and merged it in.  Thanks again to @opaldes and @jeremy-seipelt for kicking this off!

**Pull request closed without merging**